### PR TITLE
Add bloom-filter with length

### PR DESCRIPTION
**Pull request by mapleFU**
This patch add a `data_index_bloom_encoding_stats.parquet` with bloom-filter length. This file is generated using the arrow-rs:

```
cargo run --features=cli --bin parquet-rewrite -- -i data_index_bloom_encoding_stats.parquet -o data_index_bloom_encoding_with_length.parquet   --compression none --bloom-filter-enabled true --bloom-filter-ndv 2000 
```

The statistics:

```
{
  "Version": "1.0",
  "CreatedBy": "parquet-rs version 49.0.0",
  "TotalRows": "14",
  "NumberOfRowGroups": "1",
  "NumberOfRealColumns": "1",
  "NumberOfColumns": "1",
  "Columns": [
     { "Id": "0", "Name": "String", "PhysicalType": "BYTE_ARRAY", "ConvertedType": "UTF8", "LogicalType": {"Type": "String"} }
  ],
  "RowGroups": [
     {
       "Id": "0",  "TotalBytes": "199",  "TotalCompressedBytes": "199",  "Rows": "14",
       "ColumnChunks": [
          {"Id": "0", "Values": "14", "StatsSet": "True", "Stats": {" },
           "Compression": "UNCOMPRESSED", "Encodings": "PLAIN RLE RLE_DICTIONARY ", "UncompressedSize": "199", "CompressedSize": "199", BloomFilter {"offset": "253", "length": "2064"}", OffsetIndex {"offset": "2342", "length": "11"}", ColumnIndex {"offset": "2317", "length": "25"}" }
        ]
     }
  ]
}
```

The reason to re-generate a file is that, we still need to make sure the bloom-filter without length could be read.

### CODE REVIEW

**Comment by mapleFU** [comment]
@pitrou @wgtmac Would you mind take a look?

File: data/README.md [hunk 1]
```diff
@@ -28,6 +28,7 @@
 | delta_encoding_optional_column.parquet       | optional INT64 and STRING columns with delta encoding. See [delta_encoding_optional_column.md](delta_encoding_optional_column.md) for details.                   |
 | nested_structs.rust.parquet                  | Used to test that the Rust Arrow reader can lookup the correct field from a nested struct. See [ARROW-11452](https://issues.apache.org/jira/browse/ARROW-11452)  |
 | data_index_bloom_encoding_stats.parquet | optional STRING column. Contains optional metadata: bloom filters, column index, offset index and encoding stats.                                                |
+| data_index_bloom_encoding_with_length.parquet | optional STRING column. Contains optional metadata: bloom filters with bloom_filter_length, column index, offset index and encoding stats.                                                |
```
Comment (wgtmac) [inline data/README.md:31]:
```suggestion
| data_index_bloom_encoding_with_length.parquet | Same with data_index_bloom_encoding_stats.parquet but has bloom_filter_length populated in the ColumnMetaData |
```
What about this?

**Review by wgtmac — APPROVED** [review]

File: data/README.md [hunk 2]
```diff
@@ -28,6 +28,7 @@
 | delta_encoding_optional_column.parquet       | optional INT64 and STRING columns with delta encoding. See [delta_encoding_optional_column.md](delta_encoding_optional_column.md) for details.                   |
 | nested_structs.rust.parquet                  | Used to test that the Rust Arrow reader can lookup the correct field from a nested struct. See [ARROW-11452](https://issues.apache.org/jira/browse/ARROW-11452)  |
 | data_index_bloom_encoding_stats.parquet | optional STRING column. Contains optional metadata: bloom filters, column index, offset index and encoding stats.                                                |
+| data_index_bloom_encoding_with_length.parquet | Same with data_index_bloom_encoding_stats.parquet but has bloom_filter_length populated in the ColumnMetaData |
```
Comment (pitrou) [inline data/README.md:31]:
```suggestion
| data_index_bloom_encoding_with_length.parquet | Same as `data_index_bloom_encoding_stats.parquet` but has `bloom_filter_length` populated in the ColumnMetaData |
```

**Comment by pitrou** [comment]
> This patch add a `data_index_bloom_encoding_stats.parquet` with bloom-filter length. This file is generated using the arrow-rs:
> 
> ```
> cargo run --features=cli --bin parquet-rewrite -- -i /data_index_bloom_encoding_stats.parquet -o data_index_bloom_encoding_with_length.parquet   --compression gzip --bloom-filter-enabled true --bloom-filter-ndv 2000
> ```

Can you perhaps avoid using compression here?

**Comment by mapleFU** [comment]
Updated

**Review by pitrou — APPROVED** [review]

**Pull request merged**